File: STEL/Guiões/Guião 1/guiao_1.py
import sys, os
import numpy as np
import matplotlib.pyplot as plt

par_lambda = float(sys.argv[1])
print('Lambda:', par_lambda)

n_samples = int(sys.argv[2])
print('Nº de amostras:',n_samples)

#generate exponential distribuited numbers
x = np.random.exponential(1/par_lambda, n_samples) 

#histogram
#set parameters
vmin=0
vmax=5/par_lambda
delta=1/5*1/par_lambda
bin_size = round((vmax-vmin)/delta)

#plot
_ = plt.hist(x, bins=bin_size, range=(vmin,vmax)) 
plt.title("Histogram")
plt.show()

File: STEL/Guiões/Guião 1/guiao_1.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char **argv){
    
    float lambda = atof(argv[1]);
    printf("Lambda: %.1f\n", lambda);

    int n_samples = atoi(argv[2]);
    printf("Nº de amostras: %d\n", n_samples);

    double x[n_samples];
    int i,j;
    double num;

    //generate exponential distribuited numbers
    for(i=0; i<n_samples;i++){
        num=(random()+1)/(double)RAND_MAX;
        x[i]=-1/lambda*log(num);
    }


    //create histogram
    int vmin = 0;
    float vmax = 5/lambda;
    float delta = (float)1/5*1/lambda;
    int size_histo=(vmax-vmin)/delta;
    int histograma[size_histo];

    for (i=0;i<size_histo;i++){
        histograma[i]=0;
        
        for(j=0;j<n_samples;j++){
            if(x[j]>=i*delta && x[j]<(i+1)*delta){
                histograma[i]++;
            }
        }
        printf("%d ", histograma[i]);
    }
    printf("\n");

    printf("bin size: %f\n",delta);
    
    //export data

    FILE *f;
    f = fopen("samples.txt", "w");
    if (f == NULL){
        printf("Error opening file!\n");
        return 0;
    }

    for(i=0; i<size_histo;i++){
        fprintf(f, "%d ", histograma[i]);
    }

    fclose(f);

    return 0;
}